ratingcurve/plot.py

```diff
@@ -202,86 +174,89 @@ def _format_residual_plot(ax) -> None:
+    def plot(self, ax: Axes = None) -> None:
+        """Plots power-law rating curve.
 
         Parameters
         ----------
-        trace : InferenceData, optional
-            ArviZ InferenceData object containing model trace.
         ax : Axes, optional
             Pre-defined matplotlib axes.
         """
-        if trace is None:
-            trace = self.trace
+        if self.idata is None:
+            raise AttributeError('Fitted rating curve cannot be plotted as '
+                                 'model has not been fit.')
 
         ax = self.setup_plot(ax=ax)
-        self._plot_transitions(trace, ax=ax)
-        super().plot(trace, ax=ax)
+        self._plot_transitions(ax=ax)
+        super().plot(ax=ax)
 
-    def plot_residuals(self, trace: InferenceData = None, ax = None):
+    def plot_residuals(self, ax: Axes = None) -> None:
         """Plots power-law residuals.
 
         Parameters
         ----------
-        trace : InferenceData
-            ArviZ InferenceData object containing model trace.
-        ax : Axes
+        ax : Axes, optional
             Pre-defined matplotlib axes.
         """
-        if trace is None:
-            trace = self.trace
+        if self.idata is None:
+            raise AttributeError('Fitted rating curve cannot be plotted as '
+                                 'model has not been fit.')
 
         ax = self.setup_plot(ax=ax)
-        self._plot_transitions(trace, ax=ax)
-        super().plot_residuals(trace, ax=ax)
+        self._plot_transitions(ax=ax)
+        super().plot_residuals(ax=ax)
 
-    def _plot_transitions(self, trace: InferenceData, ax):
+    def _plot_transitions(self, ax: Axes) -> None:
         """Plot power-law transitions (breakpoints).
 
         Parameters
         ----------
-        trace : InferenceData
-            ArviZ Inference data containing transition points (hs)
         ax : Axes
             Pre-defined matplotlib axes.
         """
-        hs = trace.posterior['hs']
+        hs = self.idata.posterior['hs']
 
         alpha = 0.05
         hs_u = hs.mean(dim=['chain', 'draw']).data
-        hs_lower = hs.quantile(alpha/2, dim=['chain', 'draw']).data.flatten()
-        hs_upper = hs.quantile(1 - alpha/2, dim=['chain', 'draw']).data.flatten()
-
-        [ax.axhspan(l, u, color='whitesmoke') for u, l in zip(hs_lower, hs_upper)]
-        [ax.axhline(u, color='lightgray', linestyle='dotted', linewidth=NARROW_LINE) for u in hs_u]
+        hs_lower = hs.quantile(alpha/2, dim=['chain',
+                                             'draw']).data.flatten()
+        hs_upper = hs.quantile(1 - alpha/2, dim=['chain',
+                                                 'draw']).data.flatten()
+
+        [ax.axhspan(l, u, color='whitesmoke') for u, l in zip(hs_lower,
+                                                              hs_upper)]
+        [ax.axhline(u, color='lightgray', linestyle='dotted',
+                    linewidth=NARROW_LINE) for u in hs_u]
```

Comment:
```suggestion
        [ax.axhspan(l, u, color='whitesmoke') for u, l in zip(hs_lower, hs_upper)]
        [ax.axhline(u, color='lightgray', linestyle='dotted', linewidth=NARROW_LINE) for u in hs_u]
```
I agree these lines a bit long but the splits are also awkward IMO. Let's leave this for now and focus on changes related to modelbuilder